### ISSUE TITLE
Error in getparam.m

### ISSUE THREAD

**Issue by crisinma** [u0078867/ulema-ul-analyzer#34]
Hi Davide, 

I have created a new BodyModel with new AnatCalc file, because we are doing some tests on calculation of elbow flexion extension angle. I am not using VICON, but OptiTrack. Also, I dont use a pointer, but put the markers on the anatomical body parts during the data collection. 
My steps: 
- I labeled the markers in OptiTrack and exported them only with the marker name, so that ULEMA is able to read them
- I lack two markers (as I said, it is just a test to see if the workflow works), so two of the markers on the lower arm would also be used as the anatomical body parts needed for the forearm segment (coded this on the GUI). 
- I chose that I dont use a pointer in the GUI, too, by choosing that I only have a single file for calibration (which is the case).  

I got an error on the getparam.m function, so I put the debug mode in that file, on the specific line. It seems that it struggles with converting some data, but dont know exactly what needs to be inside. I guess that there should be some data that is not exported from VICON on the c3d files, and it is required by ULEMA. Could you confirm this, please? If so, is there a way to hard code it for this purpose?

Many thanks in advance! 
Best, 
Cristina

![getparamerror](https://user-images.githubusercontent.com/11502722/196149146-05ebc595-df6d-4a85-a441-2b4fd6901b94.png)

**Comment by u0078867** [comment]
This is probably due to the current C3D reader function probably not reading anything else than legacy Vicon C3D files. Have you got any C3D from OptiTrack? I am going to try to extend the C3D reader, whenever I have time.

**Comment by crisinma** [comment]
Woow that is great! Thanks! I will send it to you by email, as I cannot upload here any c3d files. 

Many thanks again! 
Cristina

**Comment by u0078867** [comment]
I have just tried to make an error-reproducer test script, just pepelining the functions **readC3D_mhs.m** and **getparam.m**, as inside **BMimportULC3D_3_AUTO.m** but those seems to work properly for your file. Hence, the problem is not that **getparam()** does not work, but that the C3D file produced by OptiTrack is missing some (sub)sections that seem necessary to be there. Could you please indicate the file and line where the buggy **getparam** function is called? Thanks.

**Comment by crisinma** [comment]
What do you mean the file and line? I just have that error from ULEMA, but nothing else. Could you walk me through how to get which file and line are the problematic ones? Then I can pass them on :-) 
Thanks!

**Comment by u0078867** [comment]
That error in the deepest, I need to know how the function above is called, with what input parameters and where. Is there any MATLAB expert in your team perhaps?

**Comment by crisinma** [comment]
No, unfortunately not... Is the error not something that you could reproduce with the file that I sent? 
Sorry if this is taking too much time! Thanks!

**Comment by u0078867** [comment]
Ok no worries. Then I think that, to be able to reproduce the error, I need the full dataset that you are using in ULEMA.
Are the files you sent me already all? Will try to look at it next week.
